proto/stream_plan.proto

```diff
@@ -179,6 +179,8 @@ message StreamNode {
   }
   // The id for the operator.
   uint64 operator_id = 1;
+  // Table ids used by this executor.
+  repeated uint64 table_ids = 23;
```

Comment:
table_id should be part of the internal plan node instead of all stream nodes? For example, HashJoin should have two table ids, and Materialize should have one table id.

Reply:
Yes they are varadic so it's a vector. I don't do it in specific node like `HashJoin` cuz i think:
1. We may need replace `opeartor_id` in future and `operator_id` is in StreamNodes. Therefore the table_ids.
2. There might be a lot of downcasting to manipulate `table_ids`. The code might not be easy to extend to other executors (e.g. have to add this field into each plan node's proto).

Suggestions welcome

Reply:
Not all operators need table_id. Only stateful executors need them. So I think it's better to place them inside each stateful plan nodes.

Reply:
Strong point. Maybe list it as optional is also okay. Wait for some others' comments.

Pros:
* Add to all stateful plan nodes might bring extra engineering cost.
* As a feature of stateful plan, I think it worth a field in `StreamNode`.

Cons:
* More semantically clear. Not all fields have this table ids.

Seems like pk_indices is similar to this condition...

Reply:
There's also some stateful executors that use `executor_id` instead of `operator_id` as keyspace. e.g., simple agg. Anyway, there should be two things -- table id and operator id. table id is for stateful storage, and operator id is for identifying an operator in fragments.

Reply:
+1 for `table_id` in internal plan node.
1. as stated above, only stateful operators have table id and there are lots of stateless operators. it does not seem like `pk_indices` which is really a common field for all stream operators.
2. for most stateful stream operators, the number and meanings of tables are fixed, so we can store them and implicit check that. for example, for the JoinNode we can store `left_table_id` and `right_table_id`.
3. another example is `MaterializeNode`, you can see, the proto `MaterializeNode` has contained the table id.

Reply:
Got this. Will switch.